pkg/detector/eventfilter_test.go

```diff
@@ -0,0 +1,274 @@
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "nginx_v1"},
+					},
+				},
+				Status: appsv1.DeploymentStatus{},
+			},
+			newObj: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "nginx_v2"},
+					},
+				},
+				Status: appsv1.DeploymentStatus{},
+			},
+			wantChange: true,
+		},
+		{
+			name: "Only status fields changed(Deployment)",
+			oldObj: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Spec:   appsv1.DeploymentSpec{},
+				Status: appsv1.DeploymentStatus{Replicas: 3, UpdatedReplicas: 3, ReadyReplicas: 0},
+			},
+			newObj: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123498",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}, {}},
+				},
+				Spec:   appsv1.DeploymentSpec{},
+				Status: appsv1.DeploymentStatus{Replicas: 3, UpdatedReplicas: 1, ReadyReplicas: 2},
+			},
+			wantChange: false,
+		},
+		{
+			name: "Both user defined fields and status fields changed(Deployment)",
+			oldObj: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "nginx_v1"},
+					},
+				},
+				Status: appsv1.DeploymentStatus{Replicas: 3, UpdatedReplicas: 3, ReadyReplicas: 0},
+			},
+			newObj: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123498",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}, {}},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"app": "nginx_v2"},
+					},
+				},
+				Status: appsv1.DeploymentStatus{Replicas: 3, UpdatedReplicas: 1, ReadyReplicas: 2},
+			},
+			wantChange: true,
+		},
+		{
+			name: "Only user defined fields changed(ConfigMap)",
+			oldObj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Immutable:  nil,
+				Data:       map[string]string{"color": "red"},
+				BinaryData: nil,
+			},
+			newObj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					ResourceVersion: "123456",
+					ManagedFields:   []metav1.ManagedFieldsEntry{{}},
+				},
+				Immutable:  nil,
+				Data:       map[string]string{"color": "green"},
+				BinaryData: nil,
+			},
+			wantChange: true,
+		},
+		{
+			name: "Only status fields changed(ConfigMap)",
```

Comment:
I can get that you tested `Deployment`(as k8s native resource) and `Workload`(as CRD), don't know why test `ConfigMap`, it's a special resource that doesn't have a `.status`?

Reply:
Yes, I think `ConfigMap` is a resource of a `storage type`, it's different from the `running type` such as `Deployment`

Reply:
`storage type`? I don't get it.